File: src/main/java/com/gitlab/daring/image/command/structure/ShowContoursCommand.java
package com.gitlab.daring.image.command.structure;

import com.gitlab.daring.image.command.BaseCommand;
import com.gitlab.daring.image.command.CommandEnv;

import static com.gitlab.daring.image.util.ImageUtils.smat;

public class ShowContoursCommand extends BaseCommand {

    final FindContoursCommand findCmd = new FindContoursCommand();
    final FilterContoursCommand filterCmd = new FilterContoursCommand(args);
    final DrawContoursCommand drawCmd = new DrawContoursCommand();

    public ShowContoursCommand(String... args) {
        super(args);
        params.addAll(filterCmd.getParams());
    }

    @Override
    public void execute(CommandEnv env) {
        findCmd.execute(env);
        filterCmd.execute(env);
        env.mat.setTo(smat(0));
        drawCmd.execute(env);
    }

}

File: src/main/kotlin/com/gitlab/daring/image/command/structure/ShowContoursCommand.kt
package com.gitlab.daring.image.command.structure

import com.gitlab.daring.image.command.CommandEnv
import com.gitlab.daring.image.command.KBaseCommand
import com.gitlab.daring.image.opencv.setTo

internal class ShowContoursCommand(args: Array<String>) : KBaseCommand(args) {

    val findCmd = FindContoursCommand(emptyArray())
    val filterCmd = FilterContoursCommand(args)
    val drawCmd = DrawContoursCommand(emptyArray())

    init {
        params.addAll(filterCmd.params)
    }

    override fun execute(env: CommandEnv) {
        findCmd.execute(env)
        filterCmd.execute(env)
        env.mat.setTo(0)
        drawCmd.execute(env)
    }

}